hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -130,8 +151,14 @@ public HoodieData<WriteStatus> compact(
     TaskContextSupplier taskContextSupplier = table.getTaskContextSupplier();
     // if this is a MDT, set up the instant range of log reader just like regular MDT snapshot reader.
     Option<InstantRange> instantRange = CompactHelpers.getInstance().getInstantRange(metaClient);
+
+    // Broadcast necessary information.
+    Option<CompactorBroadcastManager> broadcastManagerOpt = getCompactorBroadcastManager(context);
+    if (broadcastManagerOpt.isPresent()) {
+      broadcastManagerOpt.get().prepareAndBroadcast();
```

Comment:
see. why we are broadcasting if there is no partial updates? 
can we avoid that

Reply:
yes